packages/contracts-bedrock/contracts/deployment/MigrationSystemDictator.sol

```diff
@@ -0,0 +1,178 @@
+
+    /**
+     * @notice Configures the ProxyAdmin contract.
+     */
+    function step1() external onlyOwner step(1) {
+        // Set the AddressManager in the ProxyAdmin.
+        config.globalConfig.proxyAdmin.setAddressManager(config.globalConfig.addressManager);
+
+        // Set the L1CrossDomainMessenger to the RESOLVED proxy type.
+        config.globalConfig.proxyAdmin.setProxyType(
+            config.proxyAddressConfig.l1CrossDomainMessengerProxy,
+            ProxyAdmin.ProxyType.RESOLVED
+        );
+
+        // Set the implementation name for the L1CrossDomainMessenger.
+        config.globalConfig.proxyAdmin.setImplementationName(
+            config.proxyAddressConfig.l1CrossDomainMessengerProxy,
+            "OVM_L1CrossDomainMessenger"
+        );
+
+        // Set the L1StandardBridge to the CHUGSPLASH proxy type.
+        config.globalConfig.proxyAdmin.setProxyType(
+            config.proxyAddressConfig.l1StandardBridgeProxy,
+            ProxyAdmin.ProxyType.CHUGSPLASH
+        );
+    }
+
+    /**
+     * @notice Pauses the system by shutting down the L1CrossDomainMessenger and clearing many
+     *         addresses inside the AddressManager.
+     */
+    function step2() external onlyOwner step(2) {
+        // Pause the L1CrossDomainMessenger
+        L1CrossDomainMessenger(config.proxyAddressConfig.l1CrossDomainMessengerProxy).pause();
+
+        // Remove all dead addresses from the AddressManager
+        string[18] memory deads = [
+            "Proxy__OVM_L1CrossDomainMessenger",
+            "Proxy__OVM_L1StandardBridge",
+            "OVM_CanonicalTransactionChain",
+            "OVM_L2CrossDomainMessenger",
+            "OVM_DecompressionPrecompileAddress",
+            "OVM_Sequencer",
+            "OVM_Proposer",
+            "OVM_ChainStorageContainer-CTC-batches",
+            "OVM_ChainStorageContainer-CTC-queue",
+            "OVM_CanonicalTransactionChain",
+            "OVM_StateCommitmentChain",
+            "OVM_BondManager",
+            "OVM_ExecutionManager",
+            "OVM_FraudVerifier",
+            "OVM_StateManagerFactory",
+            "OVM_StateTransitionerFactory",
+            "OVM_SafetyChecker",
+            "OVM_L1MultiMessageRelayer"
+        ];
+
+        for (uint256 i = 0; i < deads.length; i++) {
+            config.globalConfig.addressManager.setAddress(deads[i], address(0));
+        }
+    }
+
+    /**
+     * @notice Transfers system ownership to the ProxyAdmin.
+     */
+    function step3() external onlyOwner step(3) {
+        // Transfer ownership of the AddressManager to the ProxyAdmin.
+        config.globalConfig.addressManager.transferOwnership(
+            address(config.globalConfig.proxyAdmin)
+        );
+
+        // Transfer ownership of the L1StandardBridge to the ProxyAdmin.
+        L1ChugSplashProxy(payable(config.proxyAddressConfig.l1StandardBridgeProxy)).setOwner(
+            address(config.globalConfig.proxyAdmin)
```

Comment:
Are we ok with leaving the legacy storage slots set in these contracts?

Reply:
Ah yeah removing the legacy storage slots is still TODO